packages/go_router/test/go_router_test.dart

```diff
@@ -492,6 +497,39 @@ void main() {
       expect(matches, hasLength(1));
       expect(router.screenFor(matches.first).runtimeType, DummyScreen);
     });
+
+    testWidgets('Handles the Android back button correctly',
```

Comment:
can you add some more test cases for more complex scenario? e.g. A GoRoute that push to root navigator that covers a shell route underneath

Reply:
813695bc9b8922618767809f3104c4522e70dee3